sql/src/test/java/io/crate/integrationtests/AggregateExpressionIntegrationTest.java

```diff
@@ -0,0 +1,124 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.integrationtests;
+
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+import static io.crate.testing.TestingHelpers.printedTable;
+
+public class AggregateExpressionIntegrationTest extends SQLTransportIntegrationTest {
+
+    @Test
+    public void test_filter_in_aggregate_expr_with_group_by() {
+        execute("SELECT" +
+                "   y, " +
+                "   COLLECT_SET(x) FILTER (WHERE x > 3), " +
+                "   COLLECT_SET(x) FILTER (WHERE x > 2) " +
+                "FROM UNNEST(" +
+                "   [1, 3, 4, 3, 5, 4]," +
+                "   ['a', 'a', 'a', 'b', 'b', 'b']) as t(x, y) " +
+                "GROUP BY y " +
+                "ORDER BY y");
+        assertThat(printedTable(response.rows()),
+                   Matchers.is("a| [4]| [3, 4]\n" +
+                               "b| [4, 5]| [3, 4, 5]\n")
+        );
+    }
+
+    @Test
+    public void test_filter_in_aggregate_expr_with_group_by_column_with_nulls() {
+        execute("SELECT" +
+                "   y, " +
+                "   COLLECT_SET(x) FILTER (WHERE x > 3) " +
+                "FROM UNNEST(" +
+                "   [1, 4, 3, 5, 4]," +
+                "   ['a', 'a', null, null, null]) AS t(x, y) " +
+                "GROUP BY y " +
+                "ORDER BY y");
+        assertThat(printedTable(response.rows()),
+                   Matchers.is("a| [4]\n" +
+                               "NULL| [4, 5]\n")
+        );
+    }
+
+    @Test
+    public void test_filter_in_aggregate_expr_for_global_aggregate() {
+        execute("SELECT" +
+                "   COLLECT_SET(x) FILTER (WHERE x > 3), " +
+                "   collect_set(x) FILTER (WHERE x > 2) " +
+                "FROM UNNEST([1, 3, 4, 2, 5, 4]) AS t(x, y)");
+        assertThat(printedTable(response.rows()), Matchers.is("[4, 5]| [3, 4, 5]\n"));
+    }
+
+    @Test
+    public void test_filter_in_aggregate_expr_with_group_by_single_number() {
+        execute("SELECT" +
+                "   COLLECT_SET(x) FILTER (WHERE x > 1) " +
+                "FROM UNNEST(" +
+                "   [1, 2, 1, 3]," +
+                "   [1, 1, 2, 2]) AS t(x, y) " +
+                "GROUP BY y");
+        assertThat(printedTable(response.rows()),
+                   Matchers.is("[2]\n" +
+                               "[3]\n"));
+    }
+
+    @Test
+    public void test_filter_in_aggregate_expr_with_group_by_single_numeric_column_with_nulls() {
```

Comment:
same here, looks like similar execution of `test_filter_in_aggregate_expr_with_group_by_column_with_nulls`